File: src/main/kotlin/fi/aalto/cs/apluscourses/actions/OpenSubmissionNotificationAction.java
package fi.aalto.cs.apluscourses.actions;

import com.intellij.notification.Notification;
import com.intellij.notification.NotificationAction;
import com.intellij.openapi.actionSystem.AnActionEvent;
import fi.aalto.cs.apluscourses.notifications.UrlRenderingErrorNotification;
import fi.aalto.cs.apluscourses.model.exercise.SubmissionResult;
import fi.aalto.cs.apluscourses.services.Notifier;
import fi.aalto.cs.apluscourses.utils.PluginResourceBundle;
import org.jetbrains.annotations.NotNull;

public class OpenSubmissionNotificationAction extends NotificationAction {

  @NotNull
  private final SubmissionResult submissionResult;

//  @NotNull
//  private final UrlRenderer submissionRenderer;

//  public OpenSubmissionNotificationAction(@NotNull SubmissionResult submissionResult) {
//    this(submissionResult, new UrlRenderer());
//  }

  /**
   * Construct the action with the given parameters. This is mostly useful for testing.
   */
  public OpenSubmissionNotificationAction(@NotNull SubmissionResult submissionResult
//                                          @NotNull UrlRenderer submissionRenderer
  ) {
    super(PluginResourceBundle.getText("notification.OpenSubmissionNotificationAction.content"));
    this.submissionResult = submissionResult;
//    this.submissionRenderer = submissionRenderer;
  }

  @Override
  public void actionPerformed(@NotNull AnActionEvent e, @NotNull Notification notification) {
    try {
//      submissionRenderer.show(submissionResult.getHtmlUrl());
    } catch (Exception ex) {
      Notifier.Companion.notify(new UrlRenderingErrorNotification(ex), e.getProject());
    }
  }
}

File: src/main/kotlin/fi/aalto/cs/apluscourses/actions/exercise/OpenSubmissionNotificationAction.kt
package fi.aalto.cs.apluscourses.actions.exercise

import com.intellij.ide.BrowserUtil
import com.intellij.notification.Notification
import com.intellij.notification.NotificationAction
import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.components.service
import com.intellij.openapi.project.Project
import fi.aalto.cs.apluscourses.model.exercise.SubmissionResult
import fi.aalto.cs.apluscourses.MyBundle
import fi.aalto.cs.apluscourses.services.Opener

class OpenSubmissionNotificationAction(
    private val submissionResult: SubmissionResult
) : NotificationAction(MyBundle.message("notification.OpenSubmissionNotificationAction.content")) {


    override fun actionPerformed(e: AnActionEvent, notification: Notification) {
        e.project?.service<Opener>()?.openSubmission(submissionResult)
    }
}
